From 580172d4425439aac96afea21de635730eade9aa Mon Sep 17 00:00:00 2001
From: Giuseppe Scrivano <gscrivan@redhat.com>
Date: Fri, 20 Dec 2013 18:58:34 +0100
Subject: [PATCH 03/10] virt-install: force call to _set_defaults before
 creating devices

RHEL-ONLY

It is fixed upstream by the commit 6af0848fb82f65dbb5925c0927a59b6abed98de0 as
part of a bigger refactoring.

Calling "_get_xml_config()" forces all default devices (e.g. add a spice
channel when --graphics spice is used) to be created before they are used.

Closes: https://bugzilla.redhat.com/show_bug.cgi?id=853386
Signed-off-by: Giuseppe Scrivano <gscrivan@redhat.com>
---
 virtinst/Guest.py | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/virtinst/Guest.py b/virtinst/Guest.py
index dcf4c4b..2dfa266 100644
--- a/virtinst/Guest.py
+++ b/virtinst/Guest.py
@@ -1205,6 +1205,9 @@ class Guest(XMLBuilderDomain.XMLBuilderDomain):
 
         self._prepare_install(meter, dry)
         try:
+            # Force all default devices to be added.
+            self._get_xml_config()
+
             # Create devices if required (disk images, etc.)
             if not dry:
                 self._create_devices(meter)
-- 
1.8.5.3

